spec/artifact-upload.spec.js

```diff
@@ -0,0 +1,91 @@
+"use strict";
+var config, container, cryptoAsync, fileCreator, fsAsync, logger;
+
+container = require("./helpers/lib/container")();
+logger = container.resolve("logger");
+fsAsync = container.resolve("fsAsync");
+cryptoAsync = container.resolve("cryptoAsync");
+fileCreator = container.resolve("fileCreator");
+config = container.resolve("config");
+
+
+/**
+ * Figures out the md5 hash of a local file.
+ *
+ * @param {string} file
+ * @return {Promise.<string>}
+ */
+function loadLocalMd5Hash(file) {
+    var hash, promise, stream;
+
+    stream = fsAsync.createReadStream(file);
+    hash = cryptoAsync.createHash("md5");
+
+    promise = new Promise((resolve, reject) => {
+        stream.on("readable", () => {
+            const data = stream.read();
+
+            if (data) {
+                hash.update(data);
+            } else {
+                resolve(hash.digest("hex"));
+            }
+        });
+        stream.on("error", (error) => {
+            reject(error);
+        });
+    });
+
+    return promise;
+}
+
+describe("LARGE_ARTIFACT_UPLOAD", () => {
+    var originalTimeout;
+
+    beforeEach(() => {
+        originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+
+        // 30 minutes
+        jasmine.DEFAULT_TIMEOUT_INTERVAL = 30 * 60 * 1000;
```

Comment:
So that it is easier to understand and easier to tweak. Although I really only need to do this once.